intruder_alarm/src/singly/mod.rs

```diff
@@ -0,0 +1,220 @@
+            _ref_ty: PhantomData,
+        }
+    }
+
+    /// Returns the length of the list.
+    #[inline]
+    pub fn len(&self) -> usize {
+        self.len
+    }
+
+    /// Returns true if the list is empty, false otherwise.
+    #[inline]
+    pub fn is_empty(&self) -> bool {
+        self.len == 0
+    }
+
+    /// Borrows the first node of the list as an `Option`.
+    ///
+    /// Note that this is distinct from `front`: this method
+    /// borrows the head _node_, not the head _element_.
+    ///
+    /// # Returns
+    ///   - `Some(&N)` if the list has elements
+    ///   - `None` if the list is empty.
+    #[inline]
+    pub fn head(&self) -> Option<&Node> {
+        self.head.as_ref()
+    }
+
+    /// Mutably borrows the head node of the list as an `Option`
+    ///
+    /// # Returns
+    ///   - `Some(&mut Node)` if the list has elements
+    ///   - `None` if the list is empty.
+    #[inline]
+    pub fn head_mut(&mut self) -> Option<&mut Node> {
+        self.head.as_mut()
+    }
+}
+
+impl<T, Node, Ref> List<T, Node, Ref>
+where
+    Node: Linked,
+    Ref: OwningRef<Node>,
+    Ref: DerefMut,
+{
+    /// Push a node to the list.
+    pub fn push(&mut self, mut node: Ref) -> &mut Self {
+        unsafe {
+            *node.next_mut() = self.head;
+            let node = Link::from_owning_ref(node);
+            self.head = node;
+            self.len += 1;
+        };
+        self
+    }
+}
+
+impl<T, Node, Ref> List<T, Node, Ref>
+where
+    Node: Linked,
+    Ref: OwningRef<Node>,
+{
+    /// Pop a node from the list.
+    pub fn pop(&mut self) -> Option<Ref> {
+        unsafe {
+            self.head.as_ptr().map(|node| {
+                self.head = (*node).take_next();
+                self.len -= 1;
+                Ref::from_ptr(node as *const Node)
+            })
+        }
+    }
+}
+
+impl<T, Node, R> List<T, Node, R>
+where
+    Node: AsRef<T>,
+{
+    /// Borrows the head item of the list as an `Option`
+    ///
+    /// # Returns
+    ///   - `Some(&T)` if the list has elements
+    ///   - `None` if the list is empty.
+    #[inline]
+    pub fn front(&self) -> Option<&T> {
+        self.head().map(Node::as_ref)
+    }
+}
+
+impl<T, Node, R> List<T, Node, R>
+where
+    Node: AsMut<T>,
+{
+    /// Mutably borrows the head element of the list as an `Option`
+    ///
+    /// # Returns
+    ///   - `Some(&mut T)` if the list has elements
+    ///   - `None` if the list is empty.
+    #[inline]
+    pub fn front_mut(&mut self) -> Option<&mut T> {
+        self.head_mut().map(Node::as_mut)
+    }
+}
+
+#[cfg(all(
+    feature = "alloc",
+    not(any(feature = "std", test))
+))]
+use alloc::boxed::Box;
+#[cfg(any(feature = "std", test))]
+use std::boxed::Box;
+
+
+#[cfg(any(feature = "alloc", feature = "std", test))]
+impl<T, Node> List<T, Node, Box<Node>>
+where
+    Node: From<T>,
+    Node: Linked,
+{
+    /// Push an item to the list.
+    #[inline]
+    pub fn push_item(&mut self, item: T) -> &mut Self {
```

Comment:
For consistency with the doubly-linked list API, I would probably call this `push`.